src/v/datalake/translation/partition_translator.cc

```diff
@@ -0,0 +1,348 @@
+  std::chrono::milliseconds new_base) {
+    _jitter = jitter_t{new_base, translation_jitter};
+    vlog(
+      _logger.info,
+      "Iceberg translation interval reset to: {}",
+      _jitter.base_duration());
+}
+
+ss::future<> partition_translator::stop() {
+    vlog(_logger.debug, "stopping partition translator in term {}", _term);
+    auto f = _gate.close();
+    _as.request_abort();
+    co_await std::move(f);
+}
+
+kafka::offset partition_translator::min_offset_for_translation() const {
+    return model::offset_cast(_partition_proxy->local_start_offset());
+}
+
+std::optional<kafka::offset>
+partition_translator::max_offset_for_translation() const {
+    // We factor in LSO to ensure only committed transactional batches are
+    // translated.
+    auto lso = _partition_proxy->last_stable_offset();
+    if (lso.has_error()) {
+        vlog(
+          _logger.warn,
+          "unable to compute lso for translation: {}",
+          lso.error());
+        return std::nullopt;
+    }
+    return kafka::prev_offset(model::offset_cast(lso.value()));
+}
+
+ss::future<std::optional<coordinator::translated_offset_range>>
+partition_translator::do_translation_for_range(
+  retry_chain_node& parent, model::record_batch_reader rdr) {
+    // This configuration only writes a single row group per file but we limit
+    // the bytes via the reader max_bytes.
+    auto writer_factory = std::make_unique<batching_parquet_writer_factory>(
+      local_path{_writer_scratch_space},
+      "", // todo(iceberg): generate a prefix with offset information
+      max_rows_per_row_group,
+      max_bytes_per_row_group);
+    auto task = translation_task{**_cloud_io};
+    const auto& ntp = _partition->ntp();
+    auto remote_path_prefix = remote_path{
+      fmt::format("{}/{}/{}", iceberg_file_path_prefix, ntp.path(), _term)};
+    lazy_abort_source las{[this] {
+        return can_continue() ? std::nullopt
+                              : std::make_optional("translator stopping");
+    }};
+    auto result = co_await task.translate(
+      std::move(writer_factory),
+      std::move(rdr),
+      remote_path_prefix,
+      parent,
+      las);
+    if (result.has_error()) {
+        vlog(_logger.warn, "Error translating range {}", result.error());
+        co_return std::nullopt;
+    }
+    co_return std::move(result.value());
+}
+
+ss::future<partition_translator::translation_success>
+partition_translator::do_translate_once(retry_chain_node& parent_rcn) {
+    if (
+      !_partition->get_ntp_config().iceberg_enabled()
+      || !_features->local().is_active(features::feature::datalake_iceberg)) {
+        vlog(_logger.debug, "iceberg config/feature disabled, nothing to do.");
+        co_return translation_success::yes;
+    }
+    auto reconcile_result = co_await reconcile_with_coordinator();
+    if (!reconcile_result) {
+        vlog(_logger.warn, "reconciliation with coordinator failed");
+        co_return translation_success::no;
+    }
```

Comment:
ya thats what the original version of the code did, I just removed it to simplify the code since this is an RPC that is done every 10s (or possibly higher) or so per partition (with iceberg enabled), seemed like a premature optimization but very easy to add back at a later time if needed.